message.go

```diff
@@ -9,20 +9,28 @@ import (
 	"time"
 )
 
-// defaultMaxTries is the default max retries.
-const defaultMaxTries = 3
+const (
+	// defaultMaxTries is the default max retries.
+	defaultMaxTries = 3
+
+	// MaxRetries runs the task until task stop returning ErrTaskRetryable
+	MaxRetries = 15
+
+	// defaultBackOff with current tries for back off
+	defaultBackOff = time.Second * 5
+)
 
 // TaskSettings can be passed to the task with specific overrides.
 type TaskSettings struct {
-	MaxTries int    `json:"max_tries"`
-	ETA      string `json:"eta"`
+	MaxTries uint      `json:"max_tries"`
+	ETA      time.Time `json:"eta"`
```

Comment:
I think there is some confusion on `delay` vs `ETA`, 
- Delay is the delay for the task to be retried next
- ETA is the overall time a task is allowed to run.
So there needs to be two settings here one for Delay and one for ETA. Following is a visualisation of this
```
|------------------------------ETA--------------------------------------------------------------------------------------|
                                                                                    
|---delay time------|[------exec task (try 1)----]|----delay time----|[------exec task (try 2)----]|----delay time----| [ETA expired]
```

Reply:
Sure, i'll rename it to delay then 👍 
I dont see any use for ETA at the moment so

Reply:
makes sense